dali/operators/util/normal_distribution_op.h

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_OPERATORS_UTIL_NORMAL_DISTRIBUTION_OP_H_
+#define DALI_OPERATORS_UTIL_NORMAL_DISTRIBUTION_OP_H_
+
+#include <random>
+#include <vector>
+#include "dali/core/convert.h"
+#include "dali/pipeline/operator/operator.h"
+
+namespace dali {
+namespace detail {
+
+const std::string kMean = "mean";      // NOLINT
+const std::string kStddev = "stddev";  // NOLINT
+const std::string kShape = "shape";    // NOLINT
+
+const int kNumOutputs = 1;
+auto kShapeDefaultValue = std::vector<int>{-1};
+
+}  // namespace detail
+
+template<typename Backend>
+class NormalDistribution : public Operator<Backend> {
+ public:
+  ~NormalDistribution() override = default;
+
+  DISABLE_COPY_MOVE_ASSIGN(NormalDistribution);
+
+ protected:
+  explicit NormalDistribution(const OpSpec &spec) :
+          Operator<Backend>(spec),
+          shape_(spec.GetArgument<std::remove_const_t<decltype(this->shape_)>>(detail::kShape)),
+          seed_(spec.GetArgument<std::remove_const_t<decltype(this->seed_)>>("seed")),
```

Comment:
`std::remove_const_t<decltype(this->seed_)>` is a very convoluted way to say `int`. I'd say just use the type

Reply:
I wrote it here on purpose, since writing just an `int` is duplicating type specification. I just think, that it is not that hard to get what is going on in this code, while changing from one type to the other will be a lot easier